### PR TITLE
Unit Test for EmailService

### PR DESCRIPTION
**Pull request by Gokilarajan**
The unit test is to make sure if there are no exception thrown if the email service is not able to send the email.

### CODE REVIEW

File: TodoList.UnitTests/Services/EmailSenderTest.cs [hunk 1]
```diff
@@ -35,6 +35,18 @@ public async Task It_Should_Send_Email()
...
         }
 
-        // TODO: Implement Should_Not_Throw_Exception_If_Can_Not_Send_Email() test
+        [Fact]
+        public async Task Should_Not_Throw_Exception_If_Can_Not_Send_Email()
+        {
+            //Arrange
+            _sendGridClientMock
+               .Setup(client => client.SendEmailAsync(It.IsAny<SendGridMessage>(), It.IsAny<CancellationToken>()))
+               .ReturnsAsync(new Response(System.Net.HttpStatusCode.InternalServerError, _httpContentMock.Object, null));
```
Comment (ghost) [inline TodoList.UnitTests/Services/EmailSenderTest.cs]: Please add this reference (`using System.Net`) to keep the code small.

**Review by ghost — CHANGES_REQUESTED** [review]

File: TodoList.UnitTests/Services/EmailSenderTest.cs (hunk 1 above)
Comment (Gokilarajan) [reply TodoList.UnitTests/Services/EmailSenderTest.cs]: The System.Net is added as Namespace.

**Review by amoraitis — CHANGES_REQUESTED** [review]
Can you please update your working branch and resolve any conflicts?

**Comment by Gokilarajan** [comment]
> Can you please update your working branch and resolve any conflicts?

Yes it is done

File: TodoList.UnitTests/Services/EmailSenderTest.cs [hunk 2]
```diff
@@ -36,5 +37,19 @@ public async Task It_Should_Send_Email()
...
+
+        [Fact]
+        public async Task Should_Not_Throw_Exception_If_Can_Not_Send_Email()
+        {
+            //Arrange
+            _sendGridClientMock
+               .Setup(client => client.SendEmailAsync(It.IsAny<SendGridMessage>(), It.IsAny<CancellationToken>()))
+               .ReturnsAsync(new Response(HttpStatusCode.InternalServerError, _httpContentMock.Object, null));
+
+            EmailSender _emailSenderService = new EmailSender(_sendGridClientMock.Object, new SendGridMessage(),_logger);
```
Comment (amoraitis) [inline TodoList.UnitTests/Services/EmailSenderTest.cs]: If this is a local variable, can we please remove the `_`?

**Review by amoraitis — CHANGES_REQUESTED** [review]

**Review by amoraitis — APPROVED** [review]
Thank you for improving the tests!

**Pull request merged**